.github/workflows/ci.yaml

```diff
@@ -239,6 +239,31 @@ jobs:
           docker image prune -a -f
           docker volume prune -f
 
+  migration-e2e:
+    name: cockroach-enterprise-operator-migration-e2e
+    runs-on: ubuntu-latest-4-core
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.22
+
+      - name: Run E2E Test
+        run: make test/e2e/migrate
+
+      - name: Clean up Docker resources
```

Comment:
Out of curiosity, what's the benefit of this? Do images, volumes, etc live beyond the CI run or otherwise cause problems if not cleaned up? If this is useful, I think a comment about the purpose would help.